File: src/main/java/com/baulsupp/oksocial/services/fitbit/FitbitUtil.java
package com.baulsupp.oksocial.services.fitbit;

import com.google.common.collect.Sets;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Set;
import okhttp3.Request;

public class FitbitUtil {
  public static final Collection<String> SCOPES =
      Arrays.asList("activity", "heartrate", "location", "nutrition", "profile",
          "settings", "sleep", "social", "weight");

  private FitbitUtil() {
  }

  public static final Set<String> API_HOSTS =
      Collections.unmodifiableSet(Sets.newHashSet(
          "api.fitbit.com")
      );

  public static Request apiRequest(String s, Request.Builder requestBuilder) {
    return requestBuilder.url("https://api.fitbit.com" + s).build();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/fitbit/FitbitUtil.kt
package com.baulsupp.oksocial.services.fitbit

import okhttp3.Request
import java.util.*

object FitbitUtil {
    val SCOPES: Collection<String> = Arrays.asList("activity", "heartrate", "location", "nutrition", "profile",
            "settings", "sleep", "social", "weight")

    val API_HOSTS = setOf((
            "api.fitbit.com")
    )

    fun apiRequest(s: String, requestBuilder: Request.Builder): Request {
        return requestBuilder.url("https://api.fitbit.com" + s).build()
    }
}
